restcomm/restcomm.ussd/src/main/java/org/restcomm/connect/ussd/interpreter/UssdInterpreter.java

```diff
@@ -490,7 +490,17 @@ public void onReceive(final Object message) throws Exception {
                 @SuppressWarnings("unchecked")
                 final CallResponse<CallInfo> response = (CallResponse<CallInfo>) message;
                 // Check from whom is the message (initial call or outbound call) and update info accordingly
-                if (sender == ussdCall) {
+
+                if( response.get().direction().contains("inbound") ){
+                    callInfo = response.get();
+                    ussdCall.tell(new Answer(callInfo.sid()), source);
+                    }
+               if ("outbound-api".equals(response.get().direction())){
+                         outboundCallInfo = response.get();
+                         fsm.transition(message, downloadingRcml);
+                }
```

Comment:
also we can add a last else here in case direction is neither of `outbound-api`, `inbound`.
In that else we can log the direction.
This is for debug purposes to handle all remaining cases.